src/notifications/NotificationsHandler.tsx

```diff
@@ -0,0 +1,331 @@
+      if (!isLowerCaseMatch(accountAddress, data.address)) {
+        walletAddress = await wallets.switchToWalletWithAddress(data.address);
+      }
+      if (!walletAddress) {
+        return;
+      }
+      Navigation.handleAction(Routes.PROFILE_SCREEN, {});
+      const zerionTransaction = store
+        .getState()
+        .data.transactions.find(tx =>
+          isLowerCaseMatch(ethereumUtils.getHash(tx) ?? '', data.hash)
+        );
+      if (zerionTransaction) {
+        showTransactionDetailsSheet(zerionTransaction, contacts, walletAddress);
+      } else {
+        const network = ethereumUtils.getNetworkFromChainId(
+          parseInt(data.chain, 10)
+        );
+        const provider = await getProviderForNetwork(network);
+        const rpcTransaction = await provider.getTransaction(data.hash);
+
+        const transactionConfirmed =
+          rpcTransaction?.blockNumber && rpcTransaction?.blockHash;
+        if (!transactionConfirmed) {
+          return;
+        }
+
+        const newTransactionDetails: NewTransactionOrAddCashTransaction = {
+          type: mapNotificationTransactionType(data.transaction_type),
+          network,
+          hash: rpcTransaction.hash,
+          status: TransactionStatus.unknown,
+          amount: rpcTransaction.value.toString(),
+          nonce: rpcTransaction.nonce,
+          from: rpcTransaction.from,
+          to: rpcTransaction.to ?? null,
+          asset: null,
+          gasLimit: rpcTransaction.gasLimit,
+          maxFeePerGas: rpcTransaction.maxFeePerGas,
+          maxPriorityFeePerGas: rpcTransaction.maxPriorityFeePerGas,
+          gasPrice: rpcTransaction.gasPrice,
+          data: rpcTransaction.data,
+        };
+
+        const parsedTransaction = await parseNewTransaction(
+          newTransactionDetails,
+          nativeCurrency
+        );
+        const resultTransaction = { ...parsedTransaction };
+        const minedAt = Math.floor(Date.now() / 1000);
+        let receipt;
+
+        try {
+          if (rpcTransaction) {
+            receipt = await rpcTransaction.wait();
+          }
+        } catch (e: any) {
+          // https://docs.ethers.io/v5/api/providers/types/#providers-TransactionResponse
+          if (e.transaction) {
+            // if a transaction field exists, it was confirmed but failed
+            resultTransaction.status = TransactionStatus.failed;
+          } else {
+            // cancelled or replaced
+            resultTransaction.status = TransactionStatus.cancelled;
+          }
+        }
+        const status = receipt?.status || 0;
+        if (!isZero(status)) {
+          let direction = TransactionDirection.out;
+          if (
+            parsedTransaction?.from &&
+            parsedTransaction?.to &&
+            isLowerCaseMatch(parsedTransaction.from, parsedTransaction.to)
+          ) {
+            direction = TransactionDirection.self;
+          }
+          if (
```

Comment:
isn't this always true if `isLowerCaseMatch(parsedTransaction.from, parsedTransaction.to)` is also true? i feel like this needs an else if otherwise if `direction` is `TransactionDirection.self` it will be always overridden by `TransactionDirection.in`